reco_utils/docker/pyspark/Dockerfile

```diff
@@ -0,0 +1,39 @@
+# Copyright (c) Microsoft Corporation. All rights reserved.
+# Licensed under the MIT License.
+
+ARG BASE_CONTAINER=jupyter/pyspark-notebook
+FROM $BASE_CONTAINER
+
+LABEL maintainer="Microsoft Recommender Project <RecoDevTeam@service.microsoft.com>"
+
+ARG WORK_PATH=/home/$NB_USER/work
+ARG RECO_PATH=$WORK_PATH/Recommenders
+
+USER root
+
+# Clone repo
+RUN git clone https://github.com/Microsoft/Recommenders.git && \
+    cp -rf Recommenders $WORK_PATH 
+
+# Install the needed packages
+RUN conda install --quiet --yes \
+    "mock==2.0.0" \
+    "fastparquet>=0.1.6" \
+    "gitpython>=2.1.8" \
+    "numpy>=1.13.3" \
+    "pymongo>=3.6.1" \
+    "pytest>=3.6.4" \
+    "scikit-surprise>=1.0.6" \
+    "lightgbm==2.2.1" 
+
+RUN pip install \
+    "azureml-sdk[notebooks,tensorboard,contrib]==1.0.18" \
+    "idna==2.7" \
+    "memory-profiler>=0.54.0" \
+    "papermill>=0.15.0" \
+    "pydocumentdb>=2.3.3" \
+    "tqdm==4.31.1"
+
+RUN fix-permissions $CONDA_DIR && \
+    fix-permissions /home/$NB_USER && \
+    fix-permissions $RECO_PATH
```

Comment:
I think it would be nice to have our notebooks in docker, similarly to what MMLSpark guys are doing: https://github.com/Azure/mmlspark/blob/master/tools/docker/Dockerfile

Reply:
Ah currently we can directly run the notebooks in the browser after running a container of the image

Reply:
Could we not pull the dependencies from the github repo? That way we won't have to change versions in two locations.

Reply:
Will investigate this again. Thanks @Microsoft/miguelgfierro and @msalvaris 

The reason for not using our yaml is that the base image has some pip/conda dependencies installed which are duplicated in our case. Another reason is that the base image has its own kernel installation and registration style which is different from our approach...